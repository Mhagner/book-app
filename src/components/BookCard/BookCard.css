.book-card__container {
    background-color: var(--color-white);
    border-radius: 4px;
    display: flex;
    padding: 1rem;
    box-shadow: 0px 6px 24px rgba(84, 16, 95, 0.13);
    margin-bottom: 1.6rem;
    transition: all .2s;
}

.book-card__container:hover {
    transform: scale(1.05);
}

.book-card__container:not(:last-child) {
    margin-bottom: 2.4rem;
}

.book-card__book-content {
    padding: 1.4rem 0;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.book-card__image {
    margin-right: 1.8rem;
    max-width: 12rem;
}

.book-card__title {
    font-size: 1.4rem;
    color: var(--grey-dark-1);
    font-weight: 600;
}

.book-card__author {
    font-size: 1.2rem;
    color: var(--color-primary);
    font-weight: normal;
}

.book-card__description {
    font-size: 1.2rem;
    color: var(--color-secondary);
}

.book-card__image-content {
    align-items: center;
    display: flex;
}

.book-card__link {
    text-decoration: wavy;
    margin-bottom: 1.6rem;
}

@media (min-width: 1026px) {
    .book-card__container {
        margin-bottom: 0;
        padding: 1.5rem;
        width: 27rem;
        min-height: 17.6rem;
    }

    .book-card__book-content {
        padding: 0;
    }

    .book-card__image {
        max-width: 10rem;
    }

    .book-card__authors {
        height: 4.1rem;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
    }

    .book-card__title,
    .book-card__description p:nth-child(2) {
        width: 100px;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
    }
}
