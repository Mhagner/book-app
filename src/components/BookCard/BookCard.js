import React from 'react';
import './BookCard.css'

const BookCard = ({
    alt,
    title,
    image,
    pageCount,
    publisher,
    published,
    authors
}) => {

    return (
        <div data-testid="book-card" className="book-card__container">
            <div className="book-card__image-content">
                <img className="book-card__image" src={image} alt={alt} />
            </div>
            <div className="book-card__book-content">
                <div>
                    <h1 className="book-card__title">{title}</h1>
                    <div className="book-card__authors">
                        {authors && authors.map((author, index) => (
                            <p key={index} className="book-card__author">{author}</p>
                        ))}
                    </div>
                </div>
                <div className="book-card__description">
                    <p>{pageCount} p√°ginas</p>
                    <p>{publisher}</p>
                    <p>Publicado em {published}</p>
                </div>
            </div>
        </div>
    )
}

export default BookCard