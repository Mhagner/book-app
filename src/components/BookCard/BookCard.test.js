import { render } from '@testing-library/react'

import BookCard from './BookCard'

describe('BookCard component', () => {
    const renderComponent = (props) => {
        return render(<BookCard {...props} />)
    }

    it('Componente should defined', () => {
        const { getByTestId } = renderComponent()

        const element = getByTestId('book-card')

        expect(element).toBeDefined()
    })

    it('Componente should render', () => {
        const props = {
            alt: 'alt',
            title: 'title',
            image: 'img/image.svg',
            pageCount: 200,
            publisher: 'Editora test',
            published: 2021,
            authors: ['Mahilson', 'Hagnner']
        }

        const { getByTestId } = renderComponent(props)

        const element = getByTestId('book-card')

        expect(element).toMatchSnapshot()
    })

})