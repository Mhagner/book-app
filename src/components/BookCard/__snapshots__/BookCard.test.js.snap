// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard component Componente should render 1`] = `
<div
  class="book-card__container"
  data-testid="book-card"
>
  <div
    class="book-card__image-content"
  >
    <img
      alt="alt"
      class="book-card__image"
      src="img/image.svg"
    />
  </div>
  <div
    class="book-card__book-content"
  >
    <div>
      <h1
        class="book-card__title"
      >
        title
      </h1>
      <div
        class="book-card__authors"
      >
        <p
          class="book-card__author"
        >
          Mahilson
        </p>
        <p
          class="book-card__author"
        >
          Hagnner
        </p>
      </div>
    </div>
    <div
      class="book-card__description"
    >
      <p>
        200
         p√°ginas
      </p>
      <p>
        Editora test
      </p>
      <p>
        Publicado em 
        2021
      </p>
    </div>
  </div>
</div>
`;
